javascript/challenges/repeatedWord/repeated-word.js

```diff
@@ -15,32 +15,69 @@ function splitter(string, seperator) {
       cache[cacheInt] = lastWord;
     }
   }
-  console.log(cache);
   return cache;
 }
 
+// function push(string) {
+//   console.log(string);
+//   let str = splitter(string, ' ');
+//   console.log(str);
+//   let checkArr = [];
+//   let j = 0;
+//   for(let i = 0; str.length; i++) {
+//     // checkArr[j] = str[i];
+//     // j++;
+//     // checkArr[j];
+//   }
+// }
 
-function repeatedWord(string) {
-  let str = splitter(string, ' ');
-  console.log(str);
-  let j = 0;
+// function repeatedWord(string) {
+//   let str = splitter(string, ' ');
+//   console.log(str);
+//   let wordCounter = 0;
+//   for (let i = 0; i < str.length; i++) {
+//     //console.log(str[i]);
+//     let j = 0;
+//     if (str[i] === str) {
+//       console.log(str[i]);
+//       console.log(str[j]);
+//       wordCounter++;
+//       j += 1;
+//       if (wordCounter === 2) {
+//         console.log(str[i]);
+//         return str[i];
+//       }
+//     }
+//   }
+// }
+
+function stringToHash(string) {
+  var hash = 0;
 
-  let wordCounter = 0;
+  if (string.length === 0) return hash;
+
+  for (let i = 0; i < string.length; i++) {
+    let char = string.charCodeAt(i);
+    hash = ((hash << 5) - hash) + char;
+    hash = hash & hash;
+  }
+
+  return hash;
+}
+
+function repeatedWord(string) {
+  let str = splitter(string, ' ');//splitting the string into an array of ind words
+  let arr = [];//instantiate arr
   for (let i = 0; i < str.length; i++) {
-    if (str[i] === str[j]) {
+    let hashedString = (stringToHash(str[i]));//hashing the ind words and capturing the result
+    if (arr.includes(hashedString)) {//if the arr includes the hashed string
       console.log(str[i]);
-      console.log(str[j]);
-      wordCounter++;
-      console.log(wordCounter);
-      j++;
-      console.log(j);
-      console.log(`in if statement ${wordCounter}`);
-      if (wordCounter === 2) {
-        console.log(str[i]);
-        return str[i];
-      }
+      return str[i];//return the string at index
+    } else {
+      arr[i] = hashedString;//else put the words into the array at their index position
     }
   }
 }
 
-module.exports = { repeatedWord, splitter };
+
+module.exports = { repeatedWord, splitter, stringToHash };
```

Comment:
Looks great so far, but tests are currently passing. Once you have a chance to come back I will regrade for you!